text/001x-battery_improvements.md

```diff
@@ -0,0 +1,188 @@
+  * Start date: 2022-03-26
+  * Contributors: Hamish Willee <hamishwillee@gmail.com>, ...
+  * Related issues: 
+    - [16-cell support #1747](https://github.com/mavlink/mavlink/pull/1747)
+	- [common: added voltage and current multipliers to BATTERY_STATUS #233](https://github.com/ArduPilot/mavlink/pull/233)
+    - [Add MAV_BATTERY_CHARGE_STATE_CHARGING #1068](https://github.com/mavlink/mavlink/pull/1068)
+
+  
+# Summary
+
+This RFC proposes:
+- a new `BATTERY_STATUS_V2` message that has a single cumulative voltage value rather than individual cell voltage arrays.
+- a new (optional) `BATTERY_VOLTAGES` message that can be scaled to as many cells as needed, and which reports them as faults.
+
+  
+# Motivation
+
+The motivation is to:
+- reduce the memory required for battery reporting ([anecdotal evidence](https://github.com/ArduPilot/mavlink/pull/233#issuecomment-976197179) indicates cases with 15% of bandwidth on a default RF channel).
+- Provide a cleaner message and design for both implementers and consumers.
+
+The [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) has two arrays that can be used to report individual cell voltages (up to 14), or a single cumulative voltage, and which cannot be truncated.
+The vast majority of consumers are only interested in the total battery voltage, and either sum the battery cells or get the cumulative voltage.
+By separating the cell voltage reporting into a separate message the new battery status can be much smaller, and the cell voltages need only be sent if the consumer is actually interested.
+
+# Detailed Design
+
+There are three parts to the design:
+1. A more efficient status message
+2. A scalable battery voltage message.
+3. Mechanisms that allow the new messages to seamlessly coexist with the old one along with eventual deprecation of the older message.
+
+## BATTERY_STATUS_V2
+
+The proposed message is:
+
+```xml
+    <message id="???" name="BATTERY_STATUS_V2">
+      <description>Battery dynamic information. This should be streamed (nominally at 1Hz). Static battery information is sent in SMART_BATTERY_INFO.</description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
+      <field type="int16_t" name="temperature" units="cdegC" invalid="INT16_MAX">Temperature of the battery. INT16_MAX field not provided.</field>
+      <field type="uint32_t" name="voltage" units="mV" invalid="UINT32_MAX">Battery voltage (total). UINT32_MAX: field not provided.</field>
+      <field type="uint32_t" name="current" units="mA" invalid="UINT32_MAX">Battery current (through all cells/loads). UINT32_MAX: field not provided.</field>
+      <field type="int32_t" name="current_consumed" units="mAh" invalid="-1">Consumed charge (estimate). -1: field not provided.</field>
```

Comment:
@dakejahl @hendjoshsr71 Just realised `current_consumed` and  "current_remaining" have a level of ambiguity.

In the past there was no way to measure the battery _remaining_ (is this still the case?). So was done is to mandate that batteries were filled for every flight, measure the power consumed using a power monitor, and estimate the remaining power by subtraction.

Do smart batteries do the same thing - the only difference from a power monitor being that the `current_consumed` is stored across reboots etc? 
Is the system clever enough to detect partial charges etc?

If so, I think probably we just need to be clear that if using a power monitor the current consumed and current remaining will be calculated on the assumption of a fully charged battery is used on every vehicle reboot.

Does that make sense?

Reply:
>In the past there was no way to measure the battery remaining (is this still the case?)

Smart batteries absolutely know the remaining capacity in mAh

>If so, I think probably we just need to be clear that if using a power monitor the current consumed and current remaining will be calculated on the assumption of a fully charged battery is used on every vehicle reboot.

Yeah this is how Ardupilot does it, and it can be very misleading and cause a lot of problems for operators who plug in a less than full battery. With a dumb power monitor I would say don't even report the capacity_remaining, since this can give the operator a false sense that they have more remaining than they actually do. But that's just a general problem with dumb batteries... so yes we should leave some hints in the description about how to properly use the fields.

Reply:
Yup. But there isn't really a better way that I know to do it that is truly any safer for a dumb battery?  If there is I would like to know? It looks like PX4 makes the same assumption? 

Because resting voltage lookup only works if the battery is rested..... and you have a table for the battery type right (not just the overall chemistry type)?

Reply:
I have no idea. But this is what the millions of power modules out there do now. So I don't think we can avoid catering for it.